### PR TITLE
Week3

### PR DESCRIPTION
**Pull request by abdullahkanatsiz7**
homework week3 part1 solutions

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -28,26 +22,139 @@
     });
     return elem;
   }
+  function formatDate(dateString) {
```
Comment (remarcmij) [inline homework/index.js]: Place a blank line between function definitions for improved readability. (Just like you use blank lines to separate paragraphs in a piece of text).

File: homework/index.js [hunk 2]
```diff
@@ -28,26 +22,139 @@
...
+    const dateTime = new Date(dateString);
+    return dateTime.toLocaleString();
+  }
+
+  function renderError(error) {
```
Comment (remarcmij) [inline homework/index.js]: Error handling extracted to a separate function: DRY 👍

File: homework/index.js [hunk 3]
```diff
@@ -1,19 +1,13 @@
...
-    const xhr = new XMLHttpRequest();
-    xhr.open('GET', url);
-    xhr.responseType = 'json';
-    xhr.onload = () => {
-      if (xhr.status >= 200 && xhr.status <= 299) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+  function fetchJSON(url) {
+    return fetch(url).then(response => {
```
Comment (remarcmij) [inline homework/index.js]: - [x] Replace XMLHttpRequest with fetch

File: homework/index.js [hunk 4]
```diff
@@ -28,26 +22,139 @@
...
+      text: repo.name,
     });
+    addRow(table, 'Description:', repo.description);
+    addRow(table, 'Fork: ', repo.forks);
+    addRow(table, 'Updated:', formatDate(repo.updated_at));
   }
 
+  function renderContributor(contributor, contributorList) {
+    const li = createAndAppend('li', contributorList);
+    const list = createAndAppend('a', li, {
```
Comment (remarcmij) [inline homework/index.js]: You are creating an anchor tag here, not a list. Your variable name should reflect that.

File: homework/index.js [hunk 5]
```diff
@@ -28,26 +22,139 @@
...
+      text: contributor.contributions,
+    });
+    list.addEventListener('click', () => {
+      window.open(contributor.html_url, '_blank');
+    });
+    list.addEventListener('key', event => {
+      if (event.key === 'Enter') window.open(contributor.html_url, '_blank');
+    });
+  }
+  async function fetchAndRender(repo) {
```
Comment (remarcmij) [inline homework/index.js]: Place a blank line before this function definition.

File: homework/index.js [hunk 6]
```diff
@@ -28,26 +22,139 @@
...
+    try {
```
Comment (remarcmij) [inline homework/index.js]: - [x] try/catch

File: homework/index.js [hunk 7]
```diff
@@ -28,26 +22,139 @@
...
+      const contributors = await fetchJSON(repo.contributors_url);
```
Comment (remarcmij) [inline homework/index.js]: - [x] async/await

File: homework/index.js [hunk 8]
```diff
@@ -28,26 +22,139 @@
...
+        text: repo.name,
+      });
+    });
+    renderRepoDetails(repositories[select.value]);
+
+    select.addEventListener('change', () => {
+      renderRepoDetails(repositories[select.value]);
+      fetchAndRender(repositories[select.value]);
+    });
+    return repositories[select.value];
```
Comment (remarcmij) [inline homework/index.js]: What's the purpose of returning this value? At which call location is this being used?

File: homework/index.js [hunk 9]
```diff
@@ -28,26 +22,139 @@
...
+    createAndAppend('section', mainArea, {
+      class: 'contributors-container',
+      id: 'right-side',
+    });
+
+    fetchJSON(url);
+    try {
+      const repository = await fetchJSON(url);
+      startPage(repository);
+      fetchAndRender(repository[0]);
```
Comment (remarcmij) [inline homework/index.js]: Why is this `fetchAndRender` here, and not in your `startPage` function? You already have a call to `renderRepoDetails` inside `startPage`. It makes sense to let `startPage` call both functions at start-up, as it already does so in the event handler.

File: homework-classes/RepoView.js [hunk 10]
```diff
@@ -19,8 +19,30 @@
      * @param {Object} repo A repository object.
      */
     render(repo) {
-      // TODO: replace this comment and the console.log with your own code
-      console.log('RepoView', repo);
+      this.container.innerHTML = '';
+      const table = createAndAppend('table', this.container);
+      const tr1 = this.addRow(table, 'Repository:', '');
+    createAndAppend('a', tr1.lastChild, {
```
Comment (remarcmij) [inline homework-classes/RepoView.js]: Somehow your indentation got screwed up.

Comment (abdullahkanatsiz7) [reply homework-classes/RepoView.js]: Thank you so much remarcmij that all valuable feedback.

**Pull request closed without merging**